From 837231b3089f066cc7ac0985775d515acd0e4b71 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Fri, 29 Apr 2011 16:59:21 +0800
Subject: [PATCH 138/635] ENGR00142737, Video can not show in primary disaply
 after disable second display

Set color key for primary display after disable second display

Signed-off-by: xianzhong <b07117@freescale.com>
---
 libgralloc/framebuffer.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 0e170df..a889215 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -434,6 +434,15 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
                     		LOGI("Error!UNBLANK FB0 failed!\n");
                     	}
 
+                        struct mxcfb_color_key key;
+                        key.enable = 1;
+                        key.color_key = 0x00000000; // Black
+                        LOGI("MXCFB_SET_CLR_KEY");
+                        if( ioctl(m->framebuffer->fd, MXCFB_SET_CLR_KEY, &key) < 0)
+                        {
+                            LOGE("Error!MXCFB_SET_CLR_KEY for fb0");
+                        }
+
                     }
                     
                     close(ctx->sec_fp);
-- 
1.8.0

